File: app/src/main/java/omniex/nl/omniex/ui/app/splash/SplashActivity.java
package omniex.nl.omniex.ui.app.splash;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EActivity;

import nl.omniex.omniexshopping.R;
import nl.omniex.omniexshopping.ui.app.auth.StartActivity_;
import nl.omniex.omniexshopping.ui.app.main.MainMenuActivity_;
import nl.omniex.omniexshopping.ui.base.BaseActivity;
import nl.omniex.omniexshopping.utils.SharedPrefUtils;

@EActivity(R.layout.activity_splash)
public class SplashActivity extends BaseActivity<SplashView, SplashPresenter> implements SplashView {

    @AfterViews
    void getAccessToken() {
        getPresenter().resolveAccessToken();
    }

    @Override
    public void onTokenFetched() {
        finish();
        if (SharedPrefUtils.isUserLogged())
            MainMenuActivity_.intent(this).start();
        else
            StartActivity_.intent(this).start();
    }

    @Override
    public void onTokenError(String error) {
        showToastMessage(error, 1);
    }
}

File: app/src/main/java/omniex/nl/omniex/ui/app/splash/SplashActivity.kt
package omniex.nl.omniex.ui.app.splash

import omniex.nl.omniex.R
import omniex.nl.omniex.ui.app.auth.StartActivity_
import omniex.nl.omniex.ui.app.main.MainMenuActivity_
import omniex.nl.omniex.ui.base.BaseActivity
import omniex.nl.omniex.utils.SharedPrefUtils
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EActivity
import javax.inject.Inject

@EActivity(R.layout.activity_splash)
open class SplashActivity : BaseActivity<SplashView, SplashPresenter>(), SplashView {

    @Inject
    lateinit var mSharedPrefUtils: SharedPrefUtils

    @AfterViews
    internal fun getAccessToken() {
        getPresenter().resolveAccessToken()
    }

    override fun onTokenFetched() {
        finish()
        if (mSharedPrefUtils.isUserLogged())
            MainMenuActivity_.intent(this).start()
        else
            StartActivity_.intent(this).start()
    }

    override fun onTokenError(error: String) {
        showToastMessage(error, 1)
    }
}
